fdbserver/ptxn/TeamVersionTracker.actor.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * TeamVersionTracker.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbserver/ptxn/TeamVersionTracker.h"
+
+#include "fdbclient/FDBTypes.h"
+#include "flow/UnitTest.h"
+
+namespace ptxn {
+
+TeamVersionTracker::TeamVersionTracker() {}
+
+void TeamVersionTracker::addTeams(const std::vector<TeamID>& teams, Version beginVersion) {
+	if (beginVersion > maxCV) {
+		maxCV = beginVersion;
+	}
+
+	for (const auto& team : teams) {
+		versions[team] = beginVersion;
+	}
+}
+
+std::map<TeamID, Version> TeamVersionTracker::updateTeams(const std::vector<TeamID>& teams, Version commitVersion) {
+	if (commitVersion > maxCV) {
+		maxCV = commitVersion;
+	}
+
+	std::map<TeamID, Version> results;
+	for (const auto& team : teams) {
+		auto it = versions.find(team);
+		ASSERT(it != versions.end() && it->second < commitVersion);
```

Comment:
I'm assuming this is going to be hot-path, and considering that, maybe should be `ASSERT_WE_THINK` instead?